bridge/src/main/java/io/stargate/bridge/service/BridgeService.java

```diff
@@ -88,7 +94,27 @@ public void executeBatch(Batch batch, StreamObserver<Response> responseObserver)
   public void describeKeyspace(
       Schema.DescribeKeyspaceQuery request,
       StreamObserver<Schema.CqlKeyspaceDescribe> responseObserver) {
-    SchemaHandler.describeKeyspace(request, persistence, responseObserver);
+
+    String decoratedName =
+        persistence.decorateKeyspaceName(request.getKeyspaceName(), GrpcService.HEADERS_KEY.get());
+
+    Keyspace keyspace = persistence.schema().keyspace(decoratedName);
+    if (keyspace == null) {
+      responseObserver.onError(
+          Status.NOT_FOUND.withDescription("Keyspace not found").asException());
+    } else if (request.hasHash() && request.getHash().getValue() == keyspace.hashCode()) {
```

Comment:
How expensive is the `keyspace.hashCode()`? OK I see this is pre-hashed, awesome :+1: